### ISSUE TITLE
[data request] IBM Diversity in Faces Dataset (DiF)

### ISSUE THREAD

**Issue by dynamicwebpaige** [tensorflow/datasets#299]
* Name of dataset: IBM Diversity in Faces (DiF)
* URL of dataset: https://www.research.ibm.com/artificial-intelligence/trusted-ai/diversity-in-faces/
* License of dataset: [Terms of Use](https://www.research.ibm.com/artificial-intelligence/trusted-ai/diversity-in-faces/documents/dif-terms-of-use.html)

* Short description of dataset and use case(s): "The Diversity in Faces (DiF) is a large and diverse dataset that seeks to advance the study of fairness and accuracy in facial recognition technology. The first of its kind available to the global research community,DiF provides a dataset of annotations of 1 million human facial images."

Folks who would also like to see this dataset in `tensorflow/datasets`, please thumbs-up so the developers can know which requests to prioritize.

**Comment by ChanchalKumarMaji** [comment]
@cyfra @dynamicwebpaige @rsepassi , please assign this issue to me.

**Comment by Oktai15** [comment]
@dynamicwebpaige hi! where can I find this dataset? Link is not related to dataset, actually...

**Comment by yl1991** [comment]
> @dynamicwebpaige hi! where can I find this dataset? Link is not related to dataset, actually...


I could not find the dataset using the link either.@dynamicwebpaige

**Comment by magomar** [comment]
No news on this dataset? what happened? I'm very interested on it too

**Comment by timlueg** [comment]
It's only available upon request.

Correct (?) URL:
[https://research.ibm.com/artificial-intelligence/trusted-ai/diversity-in-faces/](https://research.ibm.com/artificial-intelligence/trusted-ai/diversity-in-faces/)

**Comment by Oktai15** [comment]
@timlueg this link is very weird, no info on page

**Comment by jpgard** [comment]
Has anyone had success accessing this dataset by request? I have not heard back from a request for the data after over 6 weeks, despite following the protocol listed on the dataset website.

**Comment by Merlin2013** [comment]
> Has anyone had success accessing this dataset by request? I have not heard back from a request for the data after over 6 weeks, despite following the protocol listed on the dataset website.

I got the same situation,have you received any infomation?

**Comment by jpgard** [comment]
No, the official contact listed in the paper was unresponsive to my requests, as were the authors of the paper. Sad to see that the purported main contribution of their work -- the dataset -- is actually not available, as far as I can tell.

**Comment by Merlin2013** [comment]
> No, the official contact listed in the paper was unresponsive to my requests, as were the authors of the paper. Sad to see that the purported main contribution of their work -- the dataset -- is actually not available, as far as I can tell.

the official diversity-in-faces program webpage has gone, it seems that they donot want to share it any more under the public presure of public option. I am looking for someone who has the dataset and begging for a copy, but no luck at all for now.

**Comment by Merlin2013** [comment]
> No, the official contact listed in the paper was unresponsive to my requests, as were the authors of the paper. Sad to see that the purported main contribution of their work -- the dataset -- is actually not available, as far as I can tell.

the official diversity-in-faces program webpage has gone, it seems that they donot want to share it any more under the pressure of public opinion raising by media. I am looking for someone who has the dataset and begging for a copy, but no luck at all for now.

**Comment by magomar** [comment]
Tried to get this without success. This is very unfortunate. IBM got so much publicity from this announcement and then they retired access to the dataset without any explanation or clarification. They still keep the announcement online as if this dataset was still availabl , which is not the case. This is not right.

**Comment by Aspie96** [comment]
This would be absolutely great.
I agree it's a shame how private IBM is keeping this dataset

**Comment by Aspie96** [comment]
> the official diversity-in-faces program webpage has gone, it seems that they donot want to share it any more under the pressure of public opinion raising by media.

It'd be nice if the media stopped ruining everything just one day.

**Comment by dearzhangyi** [comment]
> It's only available upon request.
> 
> Correct (?) URL:
> https://research.ibm.com/artificial-intelligence/trusted-ai/diversity-in-faces/
The official diversity-in-faces program webpage has gone

**Comment by Aspie96** [comment]
They don't actually provide it it seems, which is a shame. Somebody should rebuild this dataset IMO. It wouldn't even be too hard to build a similar one.

**Comment by jpgard** [comment]
That's correct, the dataset is not even available upon request (or has not been for around a year) -- unless someone can verify that they have received the dataset from IBM during that time. The same issue also exists with some other "diverse" facial datasets, such as Gender Shades PPB dataset afaik.

Rebuilding the dataset would be a difficult, but worthy task, and the procedure is fairly well-described in the paper...

**Comment by Aspie96** [comment]
> Rebuilding the dataset would be a difficult, but worthy task, and the procedure is fairly well-described in the paper...

Yes, it'd be hard to rebuild it.
But with shared effort building a similar one isn't that hard, depending on the purpose.

I didn't check which features the original dataset had, so I can be wrong, but here's what I am thinking.

The original dataset it comes from might still be available. If it is, one can use the same dataset to extract all faces. Original images need not to be saved, only obtained and than only faces must be retained.
This can be done as a shared effort, while each client analizes only a subset of the dataset, like a botnet.
Then, by each client, some features (such as spots in the face) can be easily obtained automatically. In fact, they have to if face alignment is required. In addition, gender could be detected fairly safely by using multiple models.
I suggest when the models disagree on gender, it should be assumed it cannot be detected, thus those images, and only those, must be manually labelled. Then, an additional model can be trained on those faces only (they are the manually labelled faces of this particular dataset, so they are the most reliable) and all other faces can be classified once more. If the new annotation disagree with the old one, these faces also have to be annotated manually.
By doing so, one can collect a huge dataset of faces which are aligned, have the original source as this dataset and are labelled by gender.

However, this effort is useful if and only if access is no longer controlled by a central authority. This has proven not to be beneficial.
For this reason, those building the dataset should waive all of their right on the dataset (if they have any at all) and make it available free for everybody to download. Storage should not be a huge issue for that.

**Comment by ItamarRocha** [comment]
That's really unfortunate. I was seeking for a copy, but I've seen it wasn't even released. What a shame,

**Comment by atg-abhishek** [comment]
Anyone know of other instances where datasets have "disappeared" without a trace?

**Comment by Electronicshelf** [comment]
IBM wake up, where is the dataset?

**Comment by Aspie96** [comment]
We shouldn't trust datasets that have a gatekeeper, EVER.

If there is a gatekeaper, it is not science.

**Comment by magomar** [comment]
Some info can be found here. https://www.nbcnews.com/tech/internet/facial-recognition-s-dirty-little-secret-millions-online-photos-scraped-n981921

**Comment by atg-abhishek** [comment]
oh @magomar do you have links for some of the pages that still advertise it?

**Comment by magomar** [comment]
The IBM Research Blog is a goog example: https://www.ibm.com/blogs/research/2019/01/diversity-in-faces/

**Comment by magomar** [comment]
> Anyone know of other instances where datasets have "disappeared" without a trace?

You may find more info on other datasets here: https://megapixels.cc/

**Comment by Aspie96** [comment]
It is really bad this happens.

**Comment by jessieyaros** [comment]
For anyone still curious, based on the public audits of facial recognition tech, IBM officially stopped all facial recognition research in Jun 2020, which I'm guessing included providing public access to this dataset. 
See:
https://www.theverge.com/2020/6/8/21284683/ibm-no-longer-general-purpose-facial-recognition-analysis-software
